File: Data Structures & Algorithms/climbing-stairs/submission-0.java
class Solution {
    public int climbStairs(int n) {
        int[] nums = new int[n+1];
        if(n==1 || n==2){
            return n;
        }
        nums[1]=1;
        nums[2]=2;
        for(int i=3;i<n+1;i++){
            nums[i] = nums[i-2]+nums[i-1];
        }
        return nums[n];
    
        
    }
}

File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
    void fun(int x,vector<int> &v,set<vector<int>> &ans,vector<int> &temp){
        if(x==v.size()){
            ans.insert(temp);
            return;
        }
        fun(x+1,v,ans,temp);
        temp.push_back(v[x]);
        fun(x+1,v,ans,temp);
        temp.pop_back();
    }
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        set<vector<int>> ans;
        vector<int> temp;
        fun(0,nums,ans,temp);
        return vector<vector<int>> (ans.begin(),ans.end());
    }
};
